### ISSUE TITLE
Query string "returnQuery=false" in sanity client fetch method

### ISSUE THREAD

**Issue by alek0712** [sanity-io/client#612]
**Describe the bug**

I'm using the sanity client inside a React Native app, it was working fine until today. My logs shows that there is a 400 error when trying to reach the sanity API and it returns the following body:

`{"error":{"description":"no query","type":"queryParseError"}}`

and the url the sanity client is trying to fetch the data have this query string added at the end:

`https://PROJECT_ID.apicdn.sanity.io/v2021-10-21/data/query/DATASET?returnQuery=false`

this is my query: 

```
export const getFeaturedRecipesQuery = `*[_type == 'recipe' && data.homeCheck == true]{
    "id": _id,
    "title": data.title,
    "description": data.description,
    "prepTime": data.prepationTime,
    "image": data.headerImage.asset._ref,
    "category": {
      "id": data.category->_id,
      "name": data.category->title
    },
  }`
 ```
and this is how I fetch the data:

```
const getRecipes = async () => {
    try {
      const recipes = await sanity_client.fetch(getFeaturedRecipesQuery)
      setRecipes(recipes)
    } catch (error) {
      console.log(error)
      Sentry.captureException(error)
    }
  }
```

**To Reproduce**

1. Follow the steps to quick start guide in the @sanity/client repo https://www.npmjs.com/package/@sanity/client#quickstart

**Expected behavior**

Return the data I'm trying to fetch with my query

**Screenshot**
<img width="406" alt="Screenshot 2024-03-01 at 2 29 58 PM" src="https://github.com/sanity-io/client/assets/133240556/9503b0da-c358-4508-a329-2e48d2295264">

**Which versions of Sanity are you using?**

@sanity/client v6.15.1

**Which versions of Node.js / npm are you running?**

Node v18.17.0 / npm 9.8.1

**Comment by rexxars** [comment]
I am having trouble reproducing this issue - could you try `@sanity/client@v6.15.3`?

Fresh react native app, `app.js`:
```tsx
import { StatusBar } from 'expo-status-bar';
import { StyleSheet, Text, View } from 'react-native';
import { useState, useEffect } from 'react';
import { createClient } from '@sanity/client';

const client = createClient({
  projectId: 'espenhov',
  dataset: 'production',
  apiVersion: '2021-03-25',
  useCdn: true
})

export default function App() {
  const [data, setData] = useState(undefined)
  useEffect(() => {
    client.fetch('*[0]._id')
      .then(res => setData(res))
      .catch(err => setData(err.message))
  }, [])
  return (
    <View style={styles.container}>
      <Text>Running query!</Text>
      <Text>Query result! {data}</Text>
      <StatusBar style="auto" />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
```

Works both on Android and web.

What version of React Native are you using?

**Comment by alek0712** [comment]
> I am having trouble reproducing this issue - could you try `@sanity/client@v6.15.3`?
> 
> Fresh react native app, `app.js`:
> 
> ```tsx
> import { StatusBar } from 'expo-status-bar';
> import { StyleSheet, Text, View } from 'react-native';
> import { useState, useEffect } from 'react';
> import { createClient } from '@sanity/client';
> 
> const client = createClient({
>   projectId: 'espenhov',
>   dataset: 'production',
>   apiVersion: '2021-03-25',
>   useCdn: true
> })
> 
> export default function App() {
>   const [data, setData] = useState(undefined)
>   useEffect(() => {
>     client.fetch('*[0]._id')
>       .then(res => setData(res))
>       .catch(err => setData(err.message))
>   }, [])
>   return (
>     <View style={styles.container}>
>       <Text>Running query!</Text>
>       <Text>Query result! {data}</Text>
>       <StatusBar style="auto" />
>     </View>
>   );
> }
> 
> const styles = StyleSheet.create({
>   container: {
>     flex: 1,
>     backgroundColor: '#fff',
>     alignItems: 'center',
>     justifyContent: 'center',
>   },
> });
> ```
> 
> Works both on Android and web.
> 
> What version of React Native are you using?

I'm using React Native 0.71.8 and @sanity/client 6.7.0

**Comment by rexxars** [comment]
Thanks - I was able to get a React Native 0.71.8 reproduction up and see your issue. I've managed to track it down to the obscure `URL` and `URLSearchParams` implementations in React Native.

For now, you should be able to work around the issue by using this polyfill: https://github.com/charpeni/react-native-url-polyfill

I've also sent a PR to address this issue in the underlying request library that our client uses: https://github.com/sanity-io/get-it/pull/351

Will get back to you once we've reviewed, tested and released a new version.

**Comment by rexxars** [comment]
Fixed in [v6.15.4](https://github.com/sanity-io/client/releases/tag/v6.15.4) - thanks for reporting!
